src/app.rs

```diff
@@ -32,8 +33,8 @@ impl Action {
 
     pub fn queue_step(&self) -> usize {
         match self {
-            Action::Skip(_) | Action::Move(_, _) => 1,
-            _ => 0,
+            Action::Skip(_) | Action::Move(_, _) | Action::Delete(_) => 1,
+            Action::Rename(_) | Action::MkDir(_) => 0,
```

Comment:
Skipping `_` as a catch-all will guarantee that we need to think about queue_step() for every new Action, which is a good thing IMO (otherwise, you'd get an implicit 0 which may be surprising).